src/functionalTest/groovy/com/autonomousapps/android/projects/AndroidProjectWithKmpDependencies.groovy

```diff
@@ -0,0 +1,141 @@
+package com.autonomousapps.android.projects
+
+import com.autonomousapps.AbstractProject
+import com.autonomousapps.kit.*
+import com.autonomousapps.model.Advice
+import com.autonomousapps.model.ProjectAdvice
+
+import static com.autonomousapps.AdviceHelper.*
+import static com.autonomousapps.kit.Dependency.*
+
+final class AndroidProjectWithKmpDependencies extends AbstractProject {
+
+  final GradleProject gradleProject
+  private final String agpVersion
+  private final String additions
+
+  AndroidProjectWithKmpDependencies(String agpVersion, String additions = '') {
+    this.agpVersion = agpVersion
+    this.additions = additions
+    this.gradleProject = build()
+  }
+
+  private GradleProject build() {
+    def builder = newGradleProjectBuilder()
+    builder.withRootProject { r ->
+      r.gradleProperties = GradleProperties.minimalAndroidProperties()
+      r.withBuildScript { bs ->
+        bs.additions = additions
+        bs.buildscript = new BuildscriptBlock(
+          Repository.DEFAULT,
+          [androidPlugin(agpVersion)]
+        )
+      }
+    }
+    builder.withAndroidSubproject('app') { s ->
+      s.manifest = AndroidManifest.app('com.example.MainApplication')
+      s.sources = sources
+      s.withBuildScript { bs ->
+        bs.plugins = [Plugin.androidAppPlugin, Plugin.kotlinAndroidPlugin]
+        bs.dependencies = [
+          kotlinStdLib('implementation'),
+          appcompat('implementation'),
+
+          // Immutable collections JVM dep that should be corrected to the canonical target
+          kotlinxImmutable('implementation', "-jvm"),
+
+          // Coroutines Test JVM dep that should be
+          // - Swapped with the core dep
+          // - Core dep should use the canonical target
+          kotlinxCoroutinesTest('implementation', "-jvm"),
+
+          // A foundation compose dependency but we only use runtime APIs
```

Comment:
when you say "runtime", do you literally mean it could be a `runtimeOnly` dependency? Just want to clarify that for myself.

Reply:
ah no, I mean it's from the `androidx.compose.runtime:runtime` artifact